cmake/SundialsBuildOptionsPre.cmake

```diff
@@ -93,7 +93,11 @@ sundials_option(SUNDIALS_LOGGING_ENABLE_MPI BOOL "${DOCSTR}" "${ENABLE_MPI}"
 # ---------------------------------------------------------------
 
 if(UNIX)
-  sundials_option(USE_GENERIC_MATH BOOL "Use generic (std-c) math libraries" ON)
+  if(CMAKE_C_STANDARD STREQUAL "90")
+    sundials_option(USE_GENERIC_MATH BOOL "Use generic double precision math functions" ON ADVANCED)
```

Comment:
Should always to default to `ON` as this controls linking `libm` and without it we rely on system math functions to be there.